deployment/network-operator/templates/mellanox.com_v1alpha1_nicclusterpolicy_cr.yaml

```diff
@@ -24,6 +24,13 @@ spec:
     image: {{ .Values.ofedDriver.image }}
     repository: {{ .Values.ofedDriver.repository }}
     version: {{ .Values.ofedDriver.version }}
+    {{- if .Values.ofedDriver.env }}
```

Comment:
in general, v1.env you use supports also valueFrom key which can be used to reference values from object or from configmap/secret

so, to allow providing those you could just put the entire env section as is:
```
{{- if .Values.ofedDriver.env }}
  env: {{ toYaml .Values.ofedDriver.env | nindent 4 }}
{{- end }}
```

note: nindent number may be wrong :)